### PR TITLE
feat: Allow disabling pod and/or container metrics for runtime input

### PR DESCRIPTION
**Pull request by shorim**
## Description

Changes proposed in this pull request (what was done and why):

- Add `input.runtime.resources.pod.enabled` and `input.runtime.resources.pod.enabled` fields to allow selectively enabling/disabling of the pod and/or container runtime metrics
- Using CRD defaulting, both pod and container metrics are enabled by default
- Add unit tests
- Add e2e tests
- Add documentation for the new fields
- Adjust busola extensions
- Remove unused `DataPoint` field in `FilterProcessorMetrics` struct

Changes refer to particular issues, PRs or documents:

- #1183 

## Traceability
- [x] The PR is linked to a GitHub issue.
- [x] The PR has a milestone set.
- [x] The PR has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] If the change is user-facing, the documentation has been adjusted.
- [x] The feature is unit-tested.
- [x] The feature is e2e-tested.

### CODE REVIEW

**Review by skhalash — CHANGES_REQUESTED** [review]

File: apis/telemetry/v1beta1/metricpipeline_types.go [hunk 1]
```diff
@@ -82,12 +82,44 @@ type MetricPipelinePrometheusInput struct {
...
 type MetricPipelineRuntimeInput struct {
-	// If enabled, workload-related Kubernetes metrics are scraped. The default is `false`.
+	// If enabled, runtime metrics are scraped. The default is `false`.
 	Enabled bool `json:"enabled,omitempty"`
-	// Describes whether workload-related Kubernetes metrics from specific Namespaces are selected. System Namespaces are disabled by default.
+	// Describes whether runtime metrics from specific Namespaces are selected. System Namespaces are disabled by default.
 	// +optional
 	// +kubebuilder:default={exclude: {kyma-system, kube-system, istio-system, compass-system}}
 	Namespaces *MetricPipelineInputNamespaceSelector `json:"namespaces,omitempty"`
+	// Describes the Kubernetes resources for which runtime metrics are scraped.
```
[the remark this answers is not in the document]
Comment (shorim) [reply apis/telemetry/v1beta1/metricpipeline_types.go:91]: after discussion, we agreed that scraping is also suitable here and preferred also to use it to keep the terminology consistent

File: internal/otelcollector/config/metric/gateway/config_builder.go [hunk 2]
```diff
@@ -162,31 +192,42 @@ func makeServicePipelineConfig(pipeline *telemetryv1alpha1.MetricPipeline) confi
...
-		processors = append(processors, makeNamespaceFilterID(pipeline.Name, metric.InputSourcePrometheus))
+		processors = append(processors, namespaceFilterID(pipeline.Name, metric.InputSourcePrometheus))
 	}
 	if isIstioInputEnabled(input) && shouldFilterByNamespace(input.Istio.Namespaces) {
-		processors = append(processors, makeNamespaceFilterID(pipeline.Name, metric.InputSourceIstio))
+		processors = append(processors, namespaceFilterID(pipeline.Name, metric.InputSourceIstio))
 	}
 	if isOtlpInputEnabled(input) && input.Otlp != nil && shouldFilterByNamespace(input.Otlp.Namespaces) {
-		processors = append(processors, makeNamespaceFilterID(pipeline.Name, metric.InputSourceOtlp))
+		processors = append(processors, namespaceFilterID(pipeline.Name, metric.InputSourceOtlp))
```
[the remark this answers is not in the document]
Comment (shorim) [reply internal/otelcollector/config/metric/gateway/config_builder.go]: The otehr ID maker func check whether or not the filter should be added, while the `namespaceFilterID ` just formats the filter ID with a specific pattern. I renamed it to `formatNamespaceFilterID` to make the difference more visible.

File: docs/user/04-metrics.md [hunk 3]
```diff
@@ -331,7 +331,49 @@ spec:
         value: https://backend.example.com:4317
 ```
 
-The agent configures the [kubletstatsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kubeletstatsreceiver) for the metric groups `pod` and `container`. With that, [system metrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/kubeletstatsreceiver/documentation.md) related to containers and pods get collected.
+The agent configures the [kubletstatsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kubeletstatsreceiver) for the metric groups `pod` and `container`. With that, [system metrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/kubeletstatsreceiver/documentation.md) related to containers and pods get collected by default.
```
Comment (IwonaLanger) [inline docs/user/04-metrics.md:334]:
```suggestion
The agent configures the [kubletstatsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kubeletstatsreceiver) for the metric groups `pod` and `container`. With that, [system metrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/kubeletstatsreceiver/documentation.md) related to containers and Pods get collected by default.
```

File: docs/user/04-metrics.md [hunk 4]
```diff
@@ -331,7 +331,49 @@ spec:
...
+
+It is possible to selectively disable the collection of the pod or container metrics by defining the `resources` section in the `runtime` input.
```
Comment (IwonaLanger) [inline docs/user/04-metrics.md:336]:
```suggestion
It is possible to selectively disable the collection of the Pod or container metrics by defining the `resources` section in the `runtime` input.
```

File: docs/user/04-metrics.md [hunk 5]
```diff
@@ -331,7 +331,49 @@ spec:
...
+
+The following example drops the runtime pod metrics and only collects runtime container metrics:
```
Comment (IwonaLanger) [inline docs/user/04-metrics.md:338]:
```suggestion
The following example drops the runtime Pod metrics and only collects runtime container metrics:
```

File: docs/user/04-metrics.md [hunk 6]
```diff
@@ -331,7 +331,49 @@ spec:
...
+      resources:
+        pod:
+          enabled: false
+  output:
+    otlp:
+      endpoint:
+        value: https://backend.example.com:4317
+```
+
+The following example drops the runtime container metrics and only collects runtime pod metrics:
```
Comment (IwonaLanger) [inline docs/user/04-metrics.md:358]:
```suggestion
The following example drops the runtime container metrics and only collects runtime Pod metrics:
```

File: docs/user/resources/05-metricpipeline.md [hunk 7]
```diff
@@ -89,10 +89,15 @@ For details, see the [MetricPipeline specification file](https://github.com/kyma
 | **input.&#x200b;prometheus.&#x200b;namespaces.&#x200b;exclude**  | \[\]string | Exclude metrics from the specified Namespace names only. |
 | **input.&#x200b;prometheus.&#x200b;namespaces.&#x200b;include**  | \[\]string | Include metrics from the specified Namespace names only. |
 | **input.&#x200b;runtime**  | object | Configures runtime scraping. |
-| **input.&#x200b;runtime.&#x200b;enabled**  | boolean | If enabled, workload-related Kubernetes metrics are scraped. The default is `false`. |
-| **input.&#x200b;runtime.&#x200b;namespaces**  | object | Describes whether workload-related Kubernetes metrics from specific Namespaces are selected. System Namespaces are disabled by default. |
+| **input.&#x200b;runtime.&#x200b;enabled**  | boolean | If enabled, runtime metrics are scraped. The default is `false`. |
+| **input.&#x200b;runtime.&#x200b;namespaces**  | object | Describes whether runtime metrics from specific Namespaces are selected. System Namespaces are disabled by default. |
```
Comment (IwonaLanger) [inline docs/user/resources/05-metricpipeline.md:93]:
```suggestion
| **input.&#x200b;runtime.&#x200b;namespaces**  | object | Describes whether runtime metrics from specific namespaces are selected. System namespaces are disabled by default. |
```

File: docs/user/resources/05-metricpipeline.md [hunk 8]
```diff
@@ -89,10 +89,15 @@ For details, see the [MetricPipeline specification file](https://github.com/kyma
...
 | **input.&#x200b;runtime.&#x200b;namespaces.&#x200b;exclude**  | \[\]string | Exclude metrics from the specified Namespace names only. |
 | **input.&#x200b;runtime.&#x200b;namespaces.&#x200b;include**  | \[\]string | Include metrics from the specified Namespace names only. |
+| **input.&#x200b;runtime.&#x200b;resources**  | object | Describes the Kubernetes resources for which runtime metrics are scraped. |
+| **input.&#x200b;runtime.&#x200b;resources.&#x200b;container**  | object | Configures container runtime metrics scraping. By default, container runtime metrics  are scraped when runtime input is enabled. |
+| **input.&#x200b;runtime.&#x200b;resources.&#x200b;container.&#x200b;enabled**  | boolean | If enabled, the runtime metrics for the resource are scraped. |
+| **input.&#x200b;runtime.&#x200b;resources.&#x200b;pod**  | object | Configures pod runtime metrics scraping. By default, pod runtime metrics are scraped when runtime input is enabled. |
```
Comment (IwonaLanger) [inline docs/user/resources/05-metricpipeline.md:99]:
```suggestion
| **input.&#x200b;runtime.&#x200b;resources.&#x200b;pod**  | object | Configures Pod runtime metrics scraping. By default, Pod runtime metrics are scraped when runtime input is enabled. |
```

File: docs/user/04-metrics.md (hunk 3 above)
Comment (NHingerl) [inline docs/user/04-metrics.md:334]:
```suggestion
The agent configures the [kubletstatsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kubeletstatsreceiver) for the metric groups `pod` and `container`. With that, [system metrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/kubeletstatsreceiver/documentation.md) related to containers and Pods are collected by default.
```

File: docs/user/04-metrics.md [hunk 9]
```diff
@@ -331,7 +331,49 @@ spec:
         value: https://backend.example.com:4317
 ```
 
-The agent configures the [kubletstatsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kubeletstatsreceiver) for the metric groups `pod` and `container`. With that, [system metrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/kubeletstatsreceiver/documentation.md) related to containers and pods get collected.
+The agent configures the [kubletstatsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kubeletstatsreceiver) for the metric groups `pod` and `container`. With that, [system metrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/kubeletstatsreceiver/documentation.md) related to containers and Pods get collected by default.
```
Comment (NHingerl) [inline docs/user/04-metrics.md:334]:
```suggestion
The agent configures the [kubletstatsreceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/kubeletstatsreceiver) for the metric groups `pod` and `container`. With that, [system metrics](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/kubeletstatsreceiver/documentation.md) related to containers and Pods are collected by default.
```

File: docs/user/04-metrics.md [hunk 10]
```diff
@@ -331,7 +331,49 @@ spec:
...
+
+It is possible to selectively disable the collection of the Pod or container metrics by defining the `resources` section in the `runtime` input.
```
Comment (NHingerl) [inline docs/user/04-metrics.md:336]:
```suggestion
If you want to disable the collection of the Pod or container metrics, define the `resources` section in the `runtime` input.
```

**Review by NHingerl — APPROVED** [review]

**Review by skhalash — APPROVED** [review]

**Pull request merged**